player/lib/player/impl.ex

```diff
@@ -0,0 +1,123 @@
+defmodule Player.Impl do
+  
+  @invalid_msg "That's not a valid action, try again..."
+  
+  
+  # === Public API =========================================================== #
+  
+  def play() do
+    play(Blackjack.new_game())
+  end
+  
+  defp play(game) do
+    clear_screen()
+    get_next_action(Blackjack.table(game))
+  end
+  
+  
+  # ~~~ hand-in-progress "handle" action ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  
+  defp get_next_action({game, table = %{game_state: :in_progress}}) do
+    clear_screen()
+    print_hands(table)
+    get_hit_or_stand_input(game, table)
+  end
+  
```

Comment:
I'm certainly not going to complain about this. My one reservation is that the function already has two heads, and the first uses pattern matching while the second uses `cond`.

The judging part of me wants these to be parallel in form.

But don't change it.